src/main/java/org/aerogear/kryptowire/KryptowireServiceImpl.java

```diff
@@ -19,15 +21,24 @@
     private static final Logger logger = Logger.getLogger(KryptowireServiceImpl.class.getName());
 
     private String apiKey;
-
     private String apiEndpoint;
+    private HttpClient httpclient;
+
+    public KryptowireServiceImpl(String apiEndpoint, String apiKey, HttpClient httpclient) {
+        this(apiEndpoint, apiKey);
```

Comment:
This calls the other one that will call HttpClients.createDefault() unnecessarily, would you not be better to do it the other way round as suggested here https://github.com/aerogear/aerogear-kryptowire-jenkins-plugin/pull/5#discussion_r197132790